pkg/statistics/handle/util/ddl_event.go

```diff
@@ -33,18 +38,36 @@ type DDLEvent struct {
 	oldTableInfo *model.TableInfo
 	oldPartInfo  *model.PartitionInfo
 	columnInfos  []*model.ColumnInfo
+	// schemaID is the ID of the schema that the table belongs to.
+	// Used to filter out the system or memory tables.
+	schemaID int64
 	// This value is used to store the table ID during a transition.
 	// It applies when a table structure is being changed from partitioned to non-partitioned, or vice versa.
 	oldTableID int64
 	tp         model.ActionType
 }
 
+// IsMemOrSysDB checks whether the table is in the memory or system database.
+func (e *DDLEvent) IsMemOrSysDB(sctx sessionctx.Context) (bool, error) {
+	if intest.InTest {
+		intest.Assert(e.schemaID != 0, "schemaID should not be 0, please set it when creating the event")
+	}
```

Comment:
```suggestion
	intest.Assert(e.schemaID != 0, "schemaID should not be 0, please set it when creating the event")
```

If it is not in test, This function will be a empty function.

Reply:
Nice catch! Thank you!